slick/src/main/scala/slick/jdbc/PostgresProfile.scala

```diff
@@ -63,25 +63,27 @@ trait PostgresProfile extends JdbcProfile {
       override def schema = super.schema.filter(_ != "public") // remove default schema
     }
     override def createColumnBuilder(tableBuilder: TableBuilder, meta: MColumn): ColumnBuilder = new ColumnBuilder(tableBuilder, meta) {
-      val VarCharPattern = "^'(.*)'::character varying$".r
-      val TextPattern = "^'(.*)'::text".r
-      val IntPattern = "^\\((-?[0-9]*)\\)$".r
+      val NumericPattern = "^['(]?(-?[0-9]+\\.?[0-9]*)[')]?(?:::(?:numeric|bigint|integer))?".r
+      val TextPattern = "^'(.*)'::(?:bpchar|character varying|text)".r
+      val UUIDPattern = "^'(.*)'::uuid".r
       override def default = meta.columnDef.map((_,tpe)).collect{
         case ("true","Boolean")  => Some(Some(true))
         case ("false","Boolean") => Some(Some(false))
-        case (VarCharPattern(str),"String") => Some(Some(str))
         case (TextPattern(str),"String") => Some(Some(str))
-        case (IntPattern(v),"Int") => Some(Some(v.toInt))
-        case (IntPattern(v),"Long") => Some(Some(v.toLong))
-        case ("NULL::character varying","String") => Some(None)
```

Comment:
Is the NULL case still matched anywhere? We should probably add a test for this.